CMakeLists.txt

```diff
@@ -95,6 +95,8 @@ include_directories(test)
 include_directories(${PROJECT_SRC})
 file(GLOB TEST_SRC_FILES "test/*.cpp")
 
+SET(CMAKE_CXX_FLAGS "-Wall -Wno-sign-compare -rdynamic -Wunused -std=c++14 -pthread -D_GLIBCXX_USE_NANOSLEEP")
```

Comment:
it's not redundant.  The setting of the flags here are for the unit tests. I override the previous flags but add also the ```-Wno-sign-compare```.  The same "pattern" is done for pretty much all the pull requests. 

The reason for this is that now hard coded test values such as ```1``` are now interpreted as signed int. With gcc4 they were interpreted as unsigned int (unless they were negative)

So with gcc5x 
```EXPECT_EQ(value, 1);``` will fail to compile if value is unsigned int, since we treat warnings as errors.